src/nss_wrap.cc

```diff
@@ -0,0 +1,560 @@
+#include "nss_wrap.h"
+
+namespace node {
+namespace nss_wrap {
+
+using nss_module::nss_gethostbyname3_r;
+using nss_module::nss_gethostbyname4_r;
+using nss_module::nss_gethostbyaddr2_r;
+using v8::Array;
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::FunctionTemplate;
+using v8::Handle;
+using v8::HandleScope;
+using v8::Integer;
+using v8::Local;
+using v8::Object;
+using v8::String;
+using v8::Value;
+
+NSSReqWrap::NSSReqWrap(Environment* env,
+                       Local<Object> req_wrap_obj,
+                       char* host_val,
+                       int8_t host_type_val)
+    : ReqWrap(env, req_wrap_obj, AsyncWrap::PROVIDER_NSSREQWRAP),
```

Comment:
@mscdex Thanks. There's also a test in `parallel/test-asyncwrap-*` (can't recall the exact name, and doing this from phone) that tests if each provider can be triggered. This has recently landed, so may need to be rebased, and will fail unless a test case is added that triggers this provider. Just throw in the most minimal code to make this new provider appear.